### ISSUE TITLE
sql: Cannot add and drop (or drop and add) index in same transaction

### ISSUE THREAD

**Issue by apantel** [cockroachdb/cockroach#42844]
**Describe the problem**

It is not possible to add and drop an index (in either order) within the same transaction

**To Reproduce**

```root@:26257/defaultdb> begin;
drop index foo cascade;
alter table t add constraint foo unique(x);
commit;
pq: index "foo" being dropped, try again later```

```root@:26257/defaultdb> begin;
alter table t add constraint foo unique(x);
drop index foo cascade;
commit;
pq: index "foo" in the middle of being added, try again later```

Setting `sql_safe_updates = FALSE` allows this behavior.

**Comment by sergeyshaykhullin** [comment]
+1

**Comment by walter-weinmann** [comment]
Is there any progress? Committing the transaction doesn't help in my case!

**Comment by rafiss** [comment]
This is the more general description of the issue: https://github.com/cockroachdb/cockroach/issues/54562

**Comment by walter-weinmann** [comment]
Solved with #54562.

**Comment by zhenruyan** [comment]
+1
